app/controllers/floating_ip_controller.rb

```diff
@@ -14,4 +14,267 @@ def self.display_methods
   end
 
   menu_section :net
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
+    params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
+
+    @refresh_div = "main_div"
+
+    case params[:pressed]
+    when "floating_ip_tag"
+      tag("FloatingIp")
+    when 'floating_ip_delete'
+      delete_floating_ips
+    when "floating_ip_edit"
+      checked_floating_ip_id = get_checked_floating_ip_id(params)
+      javascript_redirect :action => "edit", :id => checked_floating_ip_id
+    when "floating_ip_new"
+      javascript_redirect :action => "new"
+    else
+      if !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
+        replace_gtl_main_div
+      else
+        render_flash
+      end
+    end
+  end
+
+  def cancel_action(message)
+    session[:edit] = nil
+    @breadcrumbs.pop if @breadcrumbs
+    javascript_redirect :action    => @lastaction,
+                        :id        => @floating_ip.id,
+                        :display   => session[:floating_ip_display],
+                        :flash_msg => message
+  end
+
+  def create
+    assert_privileges("floating_ip_new")
+    case params[:button]
+    when "cancel"
+      javascript_redirect :action    => 'show_list',
+                          :flash_msg => _("Add of new Floating IP was cancelled by the user")
+    when "add"
+      @floating_ip = FloatingIp.new
+      options = form_params
+      ems = ExtManagementSystem.find(options[:ems_id])
+      options.delete(:ems_id)
+      task_id = ems.create_floating_ip_queue(session[:userid], options)
+
+      add_flash(_("Floating IP creation failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.to_s}, :error) unless task_id.kind_of?(Fixnum)
```

Comment:
same, Fixnum should be Integer now (being fixed in https://github.com/ManageIQ/manageiq/pull/13191 for the rest)